### DISCUSSION TITLE
Rebase with auto squash

### DISCUSSION THREAD

**Discussion by apartridge** [Mergifyio/mergify#3954] [Ideas]
We currently use Mergify with queue action, with 
```
update_method: rebase
method: merge
```
to merge PRs into our master branch.

This gives us a linear history with merge commits. We also during the code review process push fixup commits (`git commit -a --fixup HEAD`) with fixes to comments, to facilitate easier review. When review is done and the author is ready to merge, the author squashes the fixups with `git rebase -i --autosquash` and force-pushes. It would be very valuable if it was possible to invoke auto-squash via Mergify. As of now, the user needs to checkout the branch locally and do this, which is a bit tedious.

It would be great with a queue action option that makes Mergify rebase with --autosquash. Another solution might be an option to the [rebase action](https://docs.mergify.com/actions/rebase/#rebase-action) to enable autosquash?

**Comment by sileht** [comment]
That's on our roadmap, we already track your interest in this feature.

**Reply by sileht** [reply]
This is officially supported and enabled by default.

**Comment by pythoniste** [comment]
@mergify-ci-bot rebase
